### PR TITLE
Merge master into develop

### PR DESCRIPTION
**Pull request by sphuber**
This merges in the changes of `v0.12.3`

### CODE REVIEW

**Comment by ltalirz** [comment]
Looking through the changes now...

File: aiida/manage/fixtures.py [hunk 1]
```diff
@@ -96,19 +96,54 @@ class FixtureManager(object):  # pylint: disable=too-many-public-methods,useless
...
 
-        def test_my_stuff(test_data):
+        def test_my_stuff(new_database):
             # run a test
 
+        @pytest.fixture(scope='function')
+        def new_database_with_daemon(aiida_profile):
+            # Configure and start daemon before running JobProcesses
+                #
+                # Note: While the global daemon implementation makes this necessary in aiida-core 0.12.x,
```
Comment (ltalirz) [inline aiida/manage/fixtures.py]: I think this can be removed now

File: aiida/restapi/translator/node.py [hunk 2]
```diff
@@ -606,78 +607,125 @@ def get_node_shape(ntype):
...
 
-        if qb_obj.count() > 0:
+        input_node_pks = {}
+        sent_no_of_incomings = qb_obj.count()
+
+        if sent_no_of_incomings > 0:
             for node_input in qb_obj.iterdict():
                 node = node_input['in']['*']
-                linktype = node_input['main--in']['label']
+                # add node if it is not present
```
Comment (ltalirz) [inline aiida/restapi/translator/node.py]: should this be moved a bit down before the "if" statement?

File: docs/source/datatypes/bands.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+``BandsData`` object is dedicated to store bands object of different types
+(electronic bands, phonons). In this section we describe the usage of the 
+``BandsData`` to store the (part of) electronic band structure of silicon
+and some logic behind its methods.
+
+To start working with ``BandsData`` we should import it using the
+``DataFactory`` and create an object of type ``BandsData``::
+
+    from aiida.orm import DataFactory
```
Comment (ltalirz) [inline docs/source/datatypes/bands.rst]:
```suggestion
    from aiida.plugins import DataFactory
```

File: aiida/manage/fixtures.py (hunk 1 above)
Comment (sphuber) [reply aiida/manage/fixtures.py]: The whole function `new_database_with_daemon` can be removed correct?

**Comment by sphuber** [comment]
@ltalirz was the change in the version number type of the export import necessary here? If so, why wasn't this broken in the release or `provenance_redesign`?

File: aiida/manage/fixtures.py (hunk 1 above)
Comment (ltalirz) [reply aiida/manage/fixtures.py]: yes

File: aiida/transports/plugins/test_all_plugins.py [hunk 4]
```diff
@@ -1263,6 +1263,7 @@ class TestExecuteCommandWait(unittest.TestCase):
     It also checks for escaping of the folder names.
     """
 
+    @unittest.skip('reenable when issue #2460 has been addressed')
```
[the remark this answers is not in the document]
Comment (sphuber) [reply aiida/transports/plugins/test_all_plugins.py]: Yes this was already removed in `provenance_redesign` but not yet in release which re-added it in the merge

File: docs/source/datatypes/bands.rst (hunk 3 above)
Comment (sphuber) [reply docs/source/datatypes/bands.rst]: done

File: aiida/restapi/translator/node.py (hunk 2 above)
Comment (sphuber) [reply aiida/restapi/translator/node.py]: done

**Comment by sphuber** [comment]
I have cleaned your two additional commits a bit such that they only include the changes as described in the commit message. Any other changes that are just leftovers that should have gone in the merge commit as well as the additions to address your comments in this PR, I have added to the merge commit

**Comment by ltalirz** [comment]
> @ltalirz was the change in the version number type of the export import necessary here? If so, why wasn't this broken in the release or provenance_redesign?

It was complaining that it was comparing float against string

**Comment by ltalirz** [comment]
I've also fixed the yaml problem - feel free to rebase if your heart desires it

**Comment by sphuber** [comment]
> I've also fixed the yaml problem - feel free to rebase if your heart desires it

Nope, looks great :+1: if the tests finish, I am happy to merge

**Comment by sphuber** [comment]
@ltalirz I think your last commit broke the tests, the linter started adding the unicode directives again in the `environment.yml`

**Comment by ltalirz** [comment]
I leave the magic merge to you :-)

**Pull request merged**